src/seraphis_main/contextual_enote_record_types.h

```diff
@@ -86,9 +87,32 @@ enum class SpEnoteSpentStatus : unsigned char
     SPENT_ONCHAIN
 };
 
+////
+// LegacyEnoteOriginContext
+// - info related to the transaction where a legacy enote was found
+// - note that an enote may originate off-chain in a partial tx where the tx id is unknown
+///
+struct LegacyEnoteOriginContext final
+{
+    /// block index of tx (-1 if index is unknown)
+    std::uint64_t block_index{static_cast<std::uint64_t>(-1)};
+    /// timestamp of tx's block (-1 if timestamp is unknown)
+    std::uint64_t block_timestamp{static_cast<std::uint64_t>(-1)};
+    /// tx id of the tx (0 if tx is unknown)
+    rct::key transaction_id{rct::zero()};
+    /// index of the enote in the tx's output set (-1 if index is unknown)
+    std::uint64_t enote_tx_index{static_cast<std::uint16_t>(-1)};
+    /// ledger index of the enote (-1 for indexing amount if index is unknown)
+    legacy_output_index_t legacy_enote_ledger_index{static_cast<std::uint64_t>(-1)};
```

Comment:
```suggestion
    /// ledger index of the enote (-1 for indexing amount if index is unknown)
    legacy_output_index_t legacy_enote_ledger_index{static_cast<std::uint64_t>(-1), 0};
```
I was quite confused for a second, because `legacy_output_index_t` has two fields.